### DISCUSSION TITLE
Upcoming v2.2.0 update

### DISCUSSION THREAD

**Discussion by sh-dv** [sh-dv/hat.sh#40] [Show and tell]
I would like to announce the upcoming update for hat.sh that brings many features including :

- Multiple files encryption/decryption.
- Darkmode
- Custom localization.
- And more..

Thus, i would like some people from the community to test out the upcoming version.
Live preview can be found on [dev.hat.sh](https://dev.hat.sh).
Repo can be found on the dev branch.

`note: 1 language (french) PR is still in progress. but title can bee seen on homepage.`

Testers github username will be added to the repositories' Readme under the acknowledgement section

Many thanks

sh-dv

<hr>

The following tests are needed:

1. Testing out the file picker UI, adding files, removing files, drag drop, reset files, total files number, size, dropping a directory.
2. Single file encryption/decryption. using both methods, password and keys.
3. Multiple files encryption/decryption. using both methods, password and keys.
4. Trying to decrypt multiple files that were encrypted using different passwords.
5. Trying to decrypt multiple files that were encrypted using different keys.
6. Trying to decrypt multiple files that have been encrypted with different methods (password/keys).
7. Trying to decrypt multiple files that have among them at least 1 corrupted file or not encrypred file.
8. Trying to decrypt multiple files that have among them at least 1 file that was encrypted with [v1.hat.sh](v1.hat.sh).
9. QR code generation of public key when generating a key pair (experimental)

Tests can be done on any browser, however, it's easier to test multiple files (over 15) with chrome because it has automatic downloading.

**Comment by sh-dv** [comment]
@samuel-lucas6 I would love to hear your feedback. When you've got time of course.

**Reply by samuel-lucas6** [reply]
I've just had a quick look. The colours for the dark mode look nice, but the text seems slightly less sharp, the page flashes white when refreshing or going to/from the about page, and the code sections of the documentation have a grey background. Didn't have any issues with the single/multiple file encryption/decryption using passwords in Brave. Tried wrong passwords, modifying one byte, selecting unencrypted files, etc. Little bit more delay with multiple files, although I have 'Ask where to save each file' turned on. The key generation seemed to work fine, and you're preventing someone encrypting to their public key. 

I recommend making the text capitalisation consistent and perhaps consider enforcing a minimum password length of 8 or 12 characters, which is something I'll probably do in v4 of Kryptor.

Sorry for not testing more. Looking good!

**Reply by sh-dv** [reply]
I really appreciate your feedback. Thank you very much!

The minimum password input is something i thought about many times, and i guess it is in fact the time to apply this change.

While reading the Kryptor V4 roadmap i came across this :

> `Enforce a minimum password length. This was done back when Kryptor had a GUI. This will be a breaking change.`

Just wondering, how would this be a breaking change?

**Reply by samuel-lucas6** [reply]
Glad it's useful. It's only a breaking change if someone uses a really short password :) Although the file format will end up changing too, so people will have to use an old version of Kryptor to decrypt old files anyway.
